astroquery/nrao/core.py

```diff
@@ -195,6 +199,87 @@ def _args_to_payload(self, **kwargs):
 
         return request_payload
 
+    def _login(self, username=None, store_password=False,
+               reenter_password=False):
+        """
+        Login to the NRAO archive
+
+        Parameters
+        ----------
+        username : str, optional
+            Username to the NRAO archive. If not given, it should be specified
+            in the config file.
+        store_password : bool, optional
+            Stores the password securely in your keyring. Default is False.
+        reenter_password : bool, optional
+            Asks for the password even if it is already stored in the
+            keyring. This is the way to overwrite an already stored passwork
+            on the keyring. Default is False.
+        """
+
+        # Developer notes:
+        # Login via https://my.nrao.edu/cas/login
+        # # this can be added to auto-redirect back to the query tool: ?service=https://archive.nrao.edu/archive/advquery.jsp
+
+        if username is None:
+            if self.USERNAME == "":
```

Comment:
This could also be written as `if not self.USERNAME` (would allow to replace it with `None`, `False`, ...).

However this is not critical.